projects/github-actions/repo-gardening/src/tasks/triage-issues/index.js

```diff
@@ -11,6 +13,81 @@ const sendSlackMessage = require( '../../utils/slack/send-slack-message' );
 
 /* global GitHub, WebhookPayloadIssue */
 
+/**
+ * Request a list of matching labels from Open AI that can be applied to the issue,
+ * based on the issue contents.
+ *
+ * @param {GitHub} octokit - Initialized Octokit REST client.
+ * @param {string} owner   - Repository owner.
+ * @param {string} repo    - Repository name.
+ * @param {string} title   - Issue title.
+ * @param {string} body    - Issue body.
+ *
+ * @return {Promise<Array>} Promise resolving to an array of labels to apply to the issue.
+ */
+async function fetchOpenAiLabelsSuggestions( octokit, owner, repo, title, body ) {
+	// Get all the Feature and Feature Group labels in the repo.
+	const pattern = /^(\[Feature\]|\[Feature Group\])/;
+	const repoLabels = await getAllLabels( octokit, owner, repo, pattern );
+
+	// If no labels are found, bail.
+	if ( repoLabels.length === 0 ) {
+		debug( 'triage-issues: No labels found in the repository. Aborting OpenAI request.' );
+		return [];
+	}
+
+	const prompt = `Here is the issue title. It is the most important part of the text you must analyse:
+${ title }
+
+Here is the issue body:
+${ body }
+
+Here are the existing labels and their descriptions:
+${ repoLabels.join( ', ' ) }
+
+Analyze the issue and suggest relevant labels. Rules:
+- Don't search for and add more labels if the existing labels match your initial assessment.
+- Use only existing labels provided.
+- Include 1 '[Feature Group]' label.
+- Include 1 to 3 '[Feature]' labels.
+- Briefly explain each label choice in 1 sentence.
+- Format your response as a JSON object with 'labels' and 'explanations' keys.
```

Comment:
@inaikem I tweaked your prompt a bit. I removed the "Product" and "Type" rules since those are currently not pulled in our label sync, not used in every repo, and in the case of "Type", they should already be added by the issue template. Happy to change that if you'd like though.

Reply:
Consider using OpenAI's [Structured Outputs](https://platform.openai.com/docs/guides/structured-outputs/introduction) to standardize the json response. It touts a 100% accuracy in the response format.

Reply:
While Structures Outputs may not be available for our model yet, I went with JSON Mode which seems like a good option too: c23f7ff944812b5724f525971b5b2e0ff6486fda